attribute-service-impl/src/main/java/org/hypertrace/core/attribute/service/AttributeServiceImpl.java

```diff
@@ -92,25 +98,33 @@ private Datastore initDataStore(Config config) {
 
   @Override
   public void create(AttributeCreateRequest request, StreamObserver<Empty> responseObserver) {
-    Optional<String> tenantId = RequestContext.CURRENT.get().getTenantId();
-    if (tenantId.isEmpty()) {
+    Optional<String> tenantIdOptional = RequestContext.CURRENT.get().getTenantId();
+    if (tenantIdOptional.isEmpty()) {
       responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
       return;
     }
 
+    final String tenantId = tenantIdOptional.orElseThrow();
+
     AttributeMetadataValidator.validate(request);
```

Comment:
Not sure if this is from your other change or pre-existing but this looks like a bug to me. There's no try/catch around this is there? A validation error would fail but never be reported to the caller who would just hang. We probably should wrap the whole create body in a try/catch rather than the individual bits being done.

Reply:
Good catch. Added try-catch block.